system-test/storage.ts

```diff
@@ -1235,118 +1144,72 @@ describe('storage', () => {
-                err => {
-                  if (err) {
-                    next(err);
-                    return;
-                  }
-
-                  next();
-                }
-              );
-            },
-            next => bucket.getMetadata((err: ApiError) => next(err)),
-          ],
-          err => {
-            assert.ifError(err);
-            assert.strictEqual(
-              bucket.metadata.retentionPolicy.retentionPeriod,
-              `${RETENTION_DURATION_SECONDS}`
-            );
-            done();
-          }
+        await storage.createBucket(bucket.name, {
+          retentionPolicy: {
+            retentionPeriod: RETENTION_DURATION_SECONDS,
+          },
+        });
+        await bucket.getMetadata();
+        assert.strictEqual(
+          bucket.metadata.retentionPolicy.retentionPeriod,
+          `${RETENTION_DURATION_SECONDS}`
         );
       });
 
-      it('should set a retention policy', done => {
+      it('should set a retention policy', async () => {
         const bucket = storage.bucket(generateName());
-
-        async.series<{}, Error | null>(
-          [
-            next => bucket.create(next),
-            next => bucket.setRetentionPeriod(RETENTION_DURATION_SECONDS, next),
-            next => bucket.getMetadata((err: ApiError) => next(err)),
-          ],
-          err => {
-            assert.ifError(err);
-            assert.strictEqual(
-              bucket.metadata.retentionPolicy.retentionPeriod,
-              `${RETENTION_DURATION_SECONDS}`
-            );
-            done();
-          }
+        await bucket.create();
+        await bucket.setRetentionPeriod(RETENTION_DURATION_SECONDS);
+        await bucket.getMetadata();
+        assert.strictEqual(
+          bucket.metadata.retentionPolicy.retentionPeriod,
+          `${RETENTION_DURATION_SECONDS}`
         );
       });
 
-      it('should lock the retention period', done => {
+      it('should lock the retention period', async () => {
         const bucket = storage.bucket(generateName());
+        await bucket.create();
+        await bucket.setRetentionPeriod(RETENTION_DURATION_SECONDS);
+        await bucket.getMetadata();
 
-        async.series<{}, Error | null>(
-          [
-            next => bucket.create(next),
-            next => bucket.setRetentionPeriod(RETENTION_DURATION_SECONDS, next),
-            next => bucket.getMetadata((err: ApiError) => next(err)),
-            next => {
-              bucket.lock(bucket.metadata.metageneration, next);
-            },
-            next =>
-              bucket.setRetentionPeriod(RETENTION_DURATION_SECONDS / 2, next),
-          ],
-          err => {
-            if (!err) {
-              done(new Error('Expected an error.'));
-              return;
-            }
-            assert.strictEqual((err as ApiError).code, 403);
-            done();
-          }
-        );
+        await bucket.lock(bucket.metadata.metageneration);
+        try {
```

Comment:
Please use an `assert.rejects` here so we're 100% sure the exception was thrown

Reply:
`assert.fejects` doesn't accept `async` function untill node 10.
reverted to 
```ts
bucket.setRetentionPeriod(RETENTION_DURATION_SECONDS / 2)
      .then(util.noop, err => assert.strictEqual(err.code, 403));
```
What version of Node are we supporting in `system-test`?